core/src/dht/dht_reducers.rs

```diff
@@ -231,14 +253,26 @@ fn reduce_remove_entry_inner(
         )));
     }
     // Update crud-status
-    let new_status_eav = create_crud_status_eav(latest_deleted_address, CrudStatus::DELETED);
+    let result = create_crud_status_eav(latest_deleted_address, CrudStatus::DELETED);
+    if result.is_err() {
+        return Err(HolochainError::ErrorGeneric(String::from(
+            "Could not create eav",
+        )));
+    }
+    let new_status_eav = result.expect("should unwrap eav");
     let meta_storage = &new_store.meta_storage().clone();
     let res = (*meta_storage.write().unwrap()).add_eav(&new_status_eav);
     if let Err(err) = res {
         return Err(err);
     }
     // Update crud-link
-    let crud_link_eav = create_crud_link_eav(latest_deleted_address, deletion_address);
+    let result = create_crud_link_eav(latest_deleted_address, deletion_address);
+    if result.is_err() {
+        return Err(HolochainError::ErrorGeneric(String::from(
+            "Could not create eav",
+        )));
+    }
+    let crud_link_eav = result.expect("Should create crud link");
```

Comment:
You can use the `?` operator for this.
```rust
let crud_link_eav = create_crud_link_eav(latest_deleted_address, deletion_address)
    .map_err(|_| HolochainError::ErrorGeneric(String::from("Could not create eav")))?
```

Reply:
I mentioned in heartbeat yesterday I would rather come in and do a follow up to the whole dht_reducers :), needs to be refactored so that it looks more idiomatic.